csharp/src/Google.Protobuf/Collections/MapField.cs

```diff
@@ -721,6 +741,69 @@ public void CopyTo(Array array, int index)
             }
         }
 
+        /// <summary>
+        /// A value type wrapping a <see cref="MapField{TKey, TValue}"/> to enable
+        /// efficient iteration without boxing.
+        /// </summary>
+        /// <seealso cref="MapField{TKey, TValue}.AsStructEnumerable"/>
+        public readonly struct Enumerable : IEnumerable<KeyValuePair<TKey, TValue>>
+        {
+            private readonly MapField<TKey, TValue> field;
+
+            internal Enumerable(MapField<TKey, TValue> field)
+            {
+                this.field = field;
+            }
+
+            /// <summary>
+            /// Creates a new <see cref="Enumerator"/> to iterate over the map.
+            /// </summary>
+            /// <returns>An <see cref="Enumerator"/> to iterate over the map.</returns>
+            public Enumerator GetEnumerator() => new Enumerator(field);
+
+            IEnumerator<KeyValuePair<TKey, TValue>> IEnumerable<KeyValuePair<TKey, TValue>>.GetEnumerator() => GetEnumerator();
+
+            IEnumerator IEnumerable.GetEnumerator() => GetEnumerator();
+        }
+
+        /// <summary>
+        /// A value type implementation of <see cref="IEnumerator{T}"/>,
+        /// to enable efficient iteration over a <see cref="MapField{TKey, TValue}"/> without boxing.
+        /// </summary>
+        /// <seealso cref="MapField{TKey, TValue}.AsStructEnumerable"/>
+        public struct Enumerator : IEnumerator<KeyValuePair<TKey, TValue>>
+        {
+            private LinkedList<KeyValuePair<TKey, TValue>>.Enumerator enumerator;
+
+            internal Enumerator(MapField<TKey, TValue> field)
+            {
+                this.enumerator = field.list.GetEnumerator();
+            }
+
+            /// <summary>
+            /// 
+            /// </summary>
```

Comment:
Are these `<summary>`s empty on purpose?

Reply:
Nope - had failed to save the file in Visual Studio! Should be fine now...